tests/test_s3urlcache.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+import logging
+import os
+import io
+import shutil
+import sys
+import unittest
+import threading
+import requests
+from http.server import BaseHTTPRequestHandler, HTTPServer
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))  # noqa
+sys.path.insert(0, pkg_root)  # noqa
+
+from dss import Config, BucketConfig
+from dss.config import Replica
+from tests.infra import start_verbose_logging, testmode
+from dss.storage.validator import S3UrlCache
+from dss.storage.validator import SizeLimitError
+from dss.util import networking
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+start_verbose_logging()
+
+KB = 1024
```

Comment:
Nit: To be precise, this is a Kibibyte, not a Kilobyte, so the appropriate unit abbreviation is KiB.